src/compiler/utils/Stack.js

```diff
@@ -0,0 +1,33 @@
+let Stack = function() {
+  let history = [],
+      memory = [];
+  let count = 0;
+  function current() {
+    return history.length ? history[history.length - 1] : null;
+  }
+  this.current = current;
+  this.stack = function() {
```

Comment:
Probably needs a new name - all the other method names are verbs. Maybe `getStack` or `getCurrentStack`